internal/core/subsume/vertex.go

```diff
@@ -170,11 +170,19 @@ func (s *subsumer) vertices(x, y *adt.Vertex) bool {
 			continue
 		}
 
-		s.missing = f
-		s.gt = a
-		s.lt = y
+		if b == nil {
```

Comment:
I'm unclear how this condition can ever be true: if b == nil, b will be set to a non-nil value in the block above.

Reply:
You are correct. I have removed this and also set the `s.missing` et al fields inside the block above.